v-next/hardhat/src/cli.ts

```diff
@@ -2,10 +2,12 @@
 // small file is loaded with sourcemaps enabled.
 process.setSourceMapsEnabled(true);
 
-// eslint-disable-next-line no-restricted-syntax -- Allow top-level await here
+/* eslint-disable no-restricted-syntax -- Allow top-level await here */
 const { main } = await import("./internal/cli/main.js");
+const { printErrorMessages } = await import("./internal/cli/error-handler.js");
+/* eslint-enable no-restricted-syntax */
 
 main(process.argv.slice(2)).catch((error: unknown) => {
-  console.error(error);
-  process.exitCode = 1;
+  printErrorMessages(error);
```

Comment:
I'd rather do this in `main`, so that that contains all the logic of the cli, and this file is just a bootstrap.

Also, `process.exit()` shouldn't be used, as it kills the process before giving it the opportunity to flush any i/o buffer (e.g. printing the error messages)

Reply:
Done in https://github.com/NomicFoundation/hardhat/pull/5455/commits/a0b79acec496263f4e79f8027f67c913783b808b